flink-runtime/src/main/java/org/apache/flink/runtime/executiongraph/IntermediateResult.java

```diff
@@ -175,35 +175,28 @@ int getNumParallelProducers() {
         return numParallelProducers;
     }
 
-    /**
-     * Currently, this method is only used to compute the maximum number of consumers. For dynamic
-     * graph, it should be called before adaptively deciding the downstream consumer parallelism.
-     */
-    int getConsumersParallelism() {
-        List<JobEdge> consumers = intermediateDataSet.getConsumers();
-        checkState(!consumers.isEmpty());
-
-        InternalExecutionGraphAccessor graph = getProducer().getGraph();
-        int consumersParallelism =
-                graph.getJobVertex(consumers.get(0).getTarget().getID()).getParallelism();
-        if (consumers.size() == 1) {
-            return consumersParallelism;
+    int getMaxConsumerParallelism() {
+        if (getProducer().getGraph().isDynamic()) {
```

Comment:
It seems that the dynamic graph does not require the parallelism of its consumers to be consistent for partition reuse, but the non-dynamic graph is different. What caused this difference? In other words, can we reset the operator's parallelism at the time of jobGraph compilation and before setVertexConfig instead of in the scheduler like FLINK-30684.